### PR TITLE
Fs change password

### PR DESCRIPTION
**Pull request by fszwedo**
Password reset endpoints added

1) Password reset request - sending email with link to password reset page
2) Password reset - resetting password based on authentication token passed when endpoint #1 is called
3) Password change - when user provides the old password

To test please:
1) First create a user using:
http://localhost:5000/api/users/register
fill the body below with data:
{
    "username": "",
    "name": "",
    "surname": "",
    "email": "yourmail",
    "type": "1",
     "password": "123"
}

2) try to change the password manually using
http://localhost:5000/api/users/changepassword
{
    "id": "id of created user",
    "oldPassword": "",
    "newPassword": ""
}

3) try to request for password email
http://localhost:5000/api/users/requestpasswordreset
{
    "id": "id of user to be changed"
}

4) in the console there will be your token displayed - copy it and reset the password using 
http://localhost:5000/api/users/resetpassword
{
    "id": "id of your user",
    "token": "token",
    "password": "newpw"
}

### CODE REVIEW

File: Api/CreateContainer.ts [hunk 1]
```diff
@@ -12,6 +12,12 @@ import UserService from './Src/Services/User';
 import UserController from './Src/Controllers/User';
 import userRoutes from './Src/Routes/User';
 
+import PasswordResetTokenModel from './Src/Models/PasswordResetToken';
+import PasswordService from './Src/Services/Password';
+import PasswordController from './Src/Controllers/Password';
+import PasswordRoutes from './Src/Routes/Password';
+import { Repository } from './Src/Repositories/Repository';
```
Comment (kjreczek) [inline Api/CreateContainer.ts:19]: We decided to name files like: Services/PasswordService and so on.

Comment (fszwedo) [reply Api/CreateContainer.ts:19]: changed

**Comment by fszwedo** [comment]
a bug I see is that the password reset token is not removed for the second subsequent reset request:

async requestPasswordReset(userId: mongoose.Types.ObjectId) {
        const user = await this.userRepository.getById(userId);
        if (!user) throw new Error ("User does not exist!");
        let token = await this.passwordTokenRepository.getById(userId);
        **if (token) await this.passwordTokenRepository.deleteById(userId);**

the last line seems to not do anything
I'm out of ideas why and could use another pair of eyes here :)

**Comment by JustynaSmoliga** [comment]
I have done steps 1,2,3 from your instruction. I have added two console.log into your code. 

 async requestPasswordReset(userId: mongoose.Types.ObjectId) {
        const user = await this.userRepository.getById(userId);
        if (!user) throw new Error ("User does not exist!");
        let token = await this.passwordTokenRepository.getById(userId);
        console.log("token");
        console.log(token);
        if (token) await this.passwordTokenRepository.deleteById(userId);
		
In console received "token" and null. I suppose that token is null and this.passwordTokenRepository.deleteById(userId) is never executed.

**Comment by fszwedo** [comment]
> I have done steps 1,2,3 from your instruction. I have added two console.log into your code.
> 
> async requestPasswordReset(userId: mongoose.Types.ObjectId) {
> const user = await this.userRepository.getById(userId);
> if (!user) throw new Error ("User does not exist!");
> let token = await this.passwordTokenRepository.getById(userId);
> console.log("token");
> console.log(token);
> if (token) await this.passwordTokenRepository.deleteById(userId);
> 
> In console received "token" and null. I suppose that token is null and this.passwordTokenRepository.deleteById(userId) is never executed.

@JustynaSmoliga isn't it because when youre requesting for the token for the first time then the token does not exist?

This if statement is for the case when the token does exist, for the second try the token is being logged for me - with the latest changes to the default repo deleting works fine so I think it should be ok but please confirm :)

**Review by JustynaSmoliga — APPROVED** [review]
Code looks good, everything works fine and bug is resolved. I don't have other comments.

**Comment by fszwedo** [comment]
@kjreczek I think your suggestion has been applied, can you confirm if you approve this pull request?

**Review by kjreczek — APPROVED** [review]
Yes, looks good. 👍

**Pull request merged**